src/containers/audio-selector.jsx

```diff
@@ -0,0 +1,156 @@
+
+const MIN_LENGTH = 0.01;
+const MIN_DURATION = 500;
+
+class AudioSelector extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleNewSelectionMouseDown',
+            'handleTrimStartMouseDown',
+            'handleTrimEndMouseDown',
+            'handleTrimStartMouseMove',
+            'handleTrimEndMouseMove',
+            'handleTrimStartMouseUp',
+            'handleTrimEndMouseUp',
+            'storeRef'
+        ]);
+
+        this.state = {
+            trimStart: props.trimStart,
+            trimEnd: props.trimEnd
+        };
+
+        this.clickStartTime = 0;
+    }
+    componentWillReceiveProps (newProps) {
+        if (newProps.trimStart === this.props.trimStart) return;
+        this.setState({
+            trimStart: newProps.trimStart,
+            trimEnd: newProps.trimEnd
+        });
+    }
+    clearSelection () {
+        this.props.onSetTrim(null, null);
+    }
+    handleNewSelectionMouseDown (e) {
+        this.initialX = getEventXY(e).x;
+        const {width, left} = this.containerElement.getBoundingClientRect();
+        this.initialTrimEnd = (this.initialX - left) / width;
+        this.initialTrimStart = this.initialTrimEnd;
+        this.props.onSetTrim(this.initialTrimStart, this.initialTrimEnd);
+
+        this.clickStartTime = Date.now();
+
+        window.addEventListener('mousemove', this.handleTrimEndMouseMove);
+        window.addEventListener('mouseup', this.handleTrimEndMouseUp);
+        window.addEventListener('touchmove', this.handleTrimEndMouseMove);
+        window.addEventListener('touchend', this.handleTrimEndMouseUp);
+    }
+    handleTrimStartMouseMove (e) {
+        const containerSize = this.containerElement.getBoundingClientRect().width;
+        const dx = (getEventXY(e).x - this.initialX) / containerSize;
+        const newTrim = Math.max(0, Math.min(1, this.initialTrimStart + dx));
+        if (newTrim > this.initialTrimEnd) {
+            this.setState({
+                trimStart: this.initialTrimEnd,
+                trimEnd: newTrim
+            });
+        } else {
+            this.setState({
+                trimStart: newTrim
+            });
+        }
+        e.preventDefault();
+    }
+    handleTrimEndMouseMove (e) {
+        const containerSize = this.containerElement.getBoundingClientRect().width;
+        const dx = (getEventXY(e).x - this.initialX) / containerSize;
+        const newTrim = Math.min(1, Math.max(0, this.initialTrimEnd + dx));
+        if (newTrim < this.initialTrimStart) {
+            this.setState({
+                trimStart: newTrim,
+                trimEnd: this.initialTrimStart
+            });
+        } else {
+            this.setState({
+                trimEnd: newTrim
+            });
+        }
+        e.preventDefault();
+    }
+    handleTrimStartMouseUp () {
+        window.removeEventListener('mousemove', this.handleTrimStartMouseMove);
```

Comment:
I think that we can use the `DragRecognizer` here, with a touchDragAngle of 90 and a drag threshold of 0. That will also fix issues with passive event listeners in touchmove.

Reply:
done!